### ISSUE TITLE
Missing/wrong error messages for tests that throw in one browser only

### ISSUE THREAD

**Issue by mattii** [litixsoft/karma-mocha-reporter#47]
Hello!
If a test fails only in PhantomJS, that error will not be listed by `karma-mocha-reporter`.

Here is a little test-suite to reproduce the error:

```javascript

function goodFunction() {
  return true;
}

function badFunction() {
  throw new Error('Oh no!');
}

function badFunctionFor(agent) {
  if (navigator.userAgent.indexOf(agent) !== -1) {
    throw new Error('Oh no! What\'s wrong with ' + agent);
  }
}

describe('Reproduce mocha-reporter output', function() {
  it('works in all browsers', function() {
    goodFunction();
  });

  it('fails in all browsers', function() {
    badFunction();
  });

  it('fails in Chrome', function() {
    badFunctionFor('Chrome');
  });

  it('fails in PhantomJS', function() {
    badFunctionFor('PhantomJS');
  });
});
```

And thats the output:

![bildschirmfoto 2016-03-26 um 18 02 59](https://cloud.githubusercontent.com/assets/784697/14061334/45003806-f37e-11e5-9386-85e3f0a7fb0f.png)

So you see, if all browsers fail, the report works, but if only one fails, things get weird:
`fails in Chrome` is incorrectly green, although it fails and its error is listed below *FAILED TESTS*. `fails in Phantom` is correctly red, but the error is not listed below *FAILED TESTS*.

I tried to debug it by adding some `console.log`s to `karma-mocha-reporter`. Unfortunately I'm not sufficiently aware of the inner workings of Karma's base reporter to fix it. But these two outputs might give you a hint:

Output of `console.log(result);` in function `specComplete`:
```javascript
...
{ description: 'fails in Chrome',
  id: 'spec7',
  log: [ 'Error: Oh no! What\'s wrong with Chrome in http://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20 (line 40)\nbadFunctionFor@http://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20:40:11\n@http://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20:54:5' ],
  skipped: false,
  success: false,
  suite: [ 'Reproduce mocha-reporter output' ],
  time: 0,
  executedExpectationsCount: 1 },

{ description: 'fails in PhantomJS',
  id: 'spec8',
  log: [],
  skipped: false,
  success: true,
  suite: [ 'Reproduce mocha-reporter output' ],
  time: 1,
  executedExpectationsCount: 0 }
...
```

Output of `console.log(suite);` in function `printFailures`:
```javascript
...
{ name: '\u001b[31m✖\u001b[39m fails in Chrome',
     isRoot: false,
     type: 'it',
     skipped: false,
     success: false,
     count: 2,
     failed: [ 'Chrome 49.0.2623 (Mac OS X 10.10.0)' ],
     visited:
      [ 'PhantomJS 2.1.1 (Mac OS X 0.0.0)',
        'Chrome 49.0.2623 (Mac OS X 10.10.0)' ],
     printed: true,
     log: [ 'Error: Oh no! What\'s wrong with Chrome in http://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20 (line 40)\nbadFunctionFor@http://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20:40:11\n@http://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20:54:5' ],
     assertionErrors: undefined },

{ name: '\u001b[32m✔\u001b[39m fails in PhantomJS',
     isRoot: false,
     type: 'it',
     skipped: false,
     success: true,
     count: 2,
     failed: [ 'PhantomJS 2.1.1 (Mac OS X 0.0.0)' ],
     visited:
      [ 'PhantomJS 2.1.1 (Mac OS X 0.0.0)',
        'Chrome 49.0.2623 (Mac OS X 10.10.0)' ],
     log: [ 'Error: Oh no! What\'s wrong with PhantomJS in http://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20 (line 40)\nbadFunctionFor@http://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20:40:58\nhttp://localhost:9877/base/demo/demo.spec.js?fb225905c65df1bed75ab7b02f98f89a0bf9be20:58:19' ],
     assertionErrors: undefined,
     printed: true } }
...
```

It seems like between the events `onSpecComplete` (moment of first output) and `onRunComplete` (moment of second output) something happens with `self.allResults`...

**Comment by 4kochi** [comment]
Thanks for the detailed description and the test case. From the first look I have no clue what's wrong. But in the `specComplete` function the property `success` is true for PhantomJS but I should be false.  I will check this next week.

**Comment by 4kochi** [comment]
Hey @mattii, i didn't forgot you. I am still investigating the error. So far it seems like the output of an it block is printed although not all browser have executed the it block.

**Comment by mattii** [comment]
Hi @4kochi, thanks for having a look at this. As far as I understand, `specComplete` is called on each completed test in each browser. It seems like the second browser, in this case Chrome, ist overwriting the `success` property of the test `it('fails in PhantomJS')`, that actually failed before in PhantomJS.

If I uncomment line 438 (`item.success = result.success;`), the failed PhantomJS test will be listed in the final failures list (unfortunately with a green check-sign). So the problem may be in the success-toggling logic in `specComplete`?

**Comment by 4kochi** [comment]
Indeed. Thanks for the hint. I guess it would be better to calculate the success property based on the current value and the new value. Something like `item.success = item.success && result.success`. In that case `false` stays `false`.

**Comment by 4kochi** [comment]
Released as v2.0.1

**Comment by mattii** [comment]
Thank you @4kochi!
